sqldelight-idea-plugin/src/main/kotlin/app/cash/sqldelight/intellij/ProjectService.kt

```diff
@@ -131,6 +135,25 @@ class ProjectService(val project: Project) : SqlDelightProjectService, Disposabl
       this.dialect = dialect
       MigrationParserDefinition.stubVersion++
       ApplicationManager.getApplication().runReadAction { invalidateAllFiles() }
+      ApplicationManager.getApplication().invokeLater {
+        ToolWindowManager.getInstance(project).getToolWindow("SqlDelight")?.remove()
+
+        val connectionManager = dialect.connectionManager()
+        if (connectionManager != null) {
+          ToolWindowManager.getInstance(project).registerToolWindow(
+            RegisterToolWindowTask(
+              id = "SqlDelight",
+              anchor = ToolWindowAnchor.BOTTOM,
+              contentFactory = SqlDelightToolWindowFactory(connectionManager),
```

Comment:
I guess the fact that this lambda captures and wraps it in a type that becomes a `val` means `connectionManager()` is only invoked once?

Still feels like it should be named `createConnectionManager()` or be a `val`. Its current instance-ness is opaque to me.